### PR TITLE
arrow: add version 7.0.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **arrow/7.0.0**

Try to close https://github.com/conan-io/conan-center-index/issues/5970

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
Oh, unexpected error occured again.

File: recipes/arrow/all/conanfile.py [hunk 1]
```diff
@@ -132,6 +141,26 @@ def configure(self):
         if self.options.with_llvm == False and self._with_llvm(True):
             raise ConanInvalidConfiguration("with_openssl options is required (or choose auto)")
 
+        if self._with_grpc():
+            raise ConanInvalidConfiguration("CCI has no grpc recipe (yet)")
```
Comment (SSE4) [inline recipes/arrow/all/conanfile.py:145]: I think it has

Comment (toge) [reply recipes/arrow/all/conanfile.py:145]: I forget to solve it.
Thank you.

Comment (toge) [reply recipes/arrow/all/conanfile.py]: fixed

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/arrow/all/conanfile.py [hunk 2]
```diff
@@ -132,6 +141,22 @@ def configure(self):
         if self.options.with_llvm == False and self._with_llvm(True):
             raise ConanInvalidConfiguration("with_openssl options is required (or choose auto)")
 
+        if self.options.with_backtrace:
+            raise ConanInvalidConfiguration("CCI has no backtrace recipe (yet)")
```
Comment (SSE4) [inline recipes/arrow/all/conanfile.py:145]: I think it has `libbacktrace/cci.20210118`

**Review by SSE4 — APPROVED** [review]

File: recipes/arrow/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/arrow/all/conanfile.py]: @SSE4
Thank you! 
Now this recipe supports libbacktrace.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**